SETUP.md

```diff
@@ -0,0 +1,300 @@
+# Setup
+
+This setup guide uses Docker. Our Docker container supports building the Zephyr applications, as well as flashing and debugging the boards.
+
+The supported platforms are Windows/x86_64, macOS/aarch64, Linux/x86_64. Other platforms are likely to work, but installing pyOCD might need extra steps.
+
+>> **On Windows, this guide uses Powershell as a host terminal. Please don't run the commands in WSL or Git Bash or cmd.**
+
+If you use Linux/WSL and want to proceed with the manual setup instead, please take a look at `MANUAL_SETUP.md` for more details.
+
+## Installation
+
+### 1. Install Git, Python
+
+Git is used to track and manage changes of our code. Python is used in the pyOCD debugger as well as few other scripts and tools.
+
+<details><summary>Windows</summary>
+
+Make sure `python` and `git` are installed **and are available from Powershell**.
+
+You can use [this](https://www.python.org/downloads/) link to install Python 3. Select option 'Add Python to PATH' in the installer.
+
+To install Git, use [this](https://git-scm.com/downloads/win) link. Git will be added to PATH by default.
+
+</details>
+
+<details><summary>macOS</summary>
+
+Before proceeding, make sure `python3` and `git` commands are available in the terminal.
+
+Python and git can be installed with homebrew:
+```
+brew install git
+brew install python
+```
+
+</details>
+
+<details><summary>Linux</summary>
+
+Before proceeding, make sure `python3` and `git` commands are available in the terminal.
+
+On Ubuntu:
+```
+sudo apt update
+sudo apt install python3 python3-pip
+```
+
+On Fedora:
+```
+sudo dnf install python3 git
+```
+
+</details>
+
+### 2. Clone the repository
+
+You can clone the repo using SSH or HTTPS. If you want to push commits to Github, you must use SSH.
+
+
+<details><summary>SSH</summary>
+
+In the host terminal, open a folder with development projects. Then run:
+```sh
+git clone git@github.com:utat-ss/finch-flight-software.git
+cd finch-flight-software
+```
+</details>
+
+<details><summary>HTTPS</summary>
+
+In the host terminal, open a folder with development projects. Then run:
+```sh
+git clone https://github.com/utat-ss/finch-flight-software.git
+cd finch-flight-software
+```
+
+</details>
+
+
+### 3. Install pyOCD
+
+pyOCD is a tool that facilitates the interaction between the STM32 chip and your computer.
+
+In the host terminal, run:
+
+___________
+
+<details><summary>Windows</summary>
+
+`python -m pip install -U pyocd`
+
+</details>
+
+<details><summary>macOS/Linux</summary>
+
+`python3 -m pip install -U pyocd`
+
+</details>
+
+___________
+
+
+If you have a Raspberry Pi or non-x86 Windows platform to install pyOCD on, check out [this](https://pyocd.io/docs/installing_on_non_x86) pyOCD guide for more information.
```

Comment:
I don't think we need to mention 'Raspberry Pi', 'non-x86' platform should be good enough.